kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java

```diff
@@ -67,32 +68,90 @@ public SchemaManager(
   /**
    * Create a new table in BigQuery.
    * @param table The BigQuery table to create.
-   * @param topic The Kafka topic used to determine the schema.
+   * @param records The sink records used to determine the schema.
    */
-  public void createTable(TableId table, String topic) {
-    Schema kafkaValueSchema = schemaRetriever.retrieveSchema(table, topic, KafkaSchemaRecordType.VALUE);
-    Schema kafkaKeySchema = kafkaKeyFieldName.isPresent() ? schemaRetriever.retrieveSchema(table, topic, KafkaSchemaRecordType.KEY) : null;
-    bigQuery.create(constructTableInfo(table, kafkaKeySchema, kafkaValueSchema));
+  public void createTable(TableId table, Set<SinkRecord> records) {
```

Comment:
I tried to avoid looping through the Records twice.(Once for getting the unionized schemas and again for the table description). But, this is much cleaner and readable. Will split it up the method.